charts/spire/charts/spire-server/templates/ingress.yaml

```diff
@@ -0,0 +1,87 @@
+{{- $fullName := include "spire-server.fullname" . -}}
+{{- if .Values.ingress.enabled -}}
+{{- $port := .Values.service.port -}}
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "spire-server.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- with .Values.ingress.className }}
+  ingressClassName: {{ . }}
+  {{- end }}
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
+  rules:
+    {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: {{ $port }}
+          {{- end }}
+    {{- end }}
+{{- end }}
+{{ if .Values.federation.ingress.enabled -}}
+---
+{{- $fedPort := .Values.federation.bundleEndpoint.port -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}-federation
+  labels:
+    {{- include "spire-server.labels" . | nindent 4 }}
+  {{- with .Values.federation.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- with .Values.federation.ingress.className }}
+  ingressClassName: {{ . }}
+  {{- end }}
+  {{- if .Values.federation.ingress.tls }}
+  tls:
+    {{- range .Values.federation.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
+  rules:
+    {{- range .Values.federation.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: {{ $fedPort }}
+          {{- end }}
+    {{- end }}
```

Comment:
This part can probably be a macro to be reused by both the ingresses?

Reply:
Yeah. I think I can make it a macro. So far each of them is pretty similar...

I'll take a stab at it.

Reply:
Done